fsharp-backend/src/QueueWorker/QueueWorker.fs

```diff
@@ -11,87 +13,168 @@ open LibBackend.Db
 module PT = LibExecution.ProgramTypes
 module PTParser = LibExecution.ProgramTypesParser
 module RT = LibExecution.RuntimeTypes
+module AT = LibExecution.AnalysisTypes
 module PT2RT = LibExecution.ProgramTypesToRuntimeTypes
-module EQ = LibBackend.EventQueue
+module EQ = LibBackend.EventQueueV2
 module TI = LibBackend.TraceInputs
 module Execution = LibExecution.Execution
 module Pusher = LibBackend.Pusher
 module RealExecution = LibRealExecution.RealExecution
 module Canvas = LibBackend.Canvas
+module DvalReprExternal = LibExecution.DvalReprExternal
 
 module LD = LibService.LaunchDarkly
 module Telemetry = LibService.Telemetry
 module Rollbar = LibService.Rollbar
 
-let dequeueAndProcess () : Task<Result<Option<RT.Dval>, exn>> =
+let shutdown = ref false
```

Comment:
I'd prefer `shouldShutdown`, but not a huge deal.

Reply:
Yeah, it matches the name in CronChecker too